src/main/java/com/gotocompany/depot/maxcompute/converter/type/PrimitiveTypeInfoConverter.java

```diff
@@ -0,0 +1,42 @@
+package com.gotocompany.depot.maxcompute.converter.type;
+
+import com.aliyun.odps.type.TypeInfo;
+import com.aliyun.odps.type.TypeInfoFactory;
+import com.google.protobuf.Descriptors;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class PrimitiveTypeInfoConverter implements TypeInfoConverter {
+
+    private static final Map<Descriptors.FieldDescriptor.Type, TypeInfo> PROTO_TYPE_MAP;
+    static {
+        PROTO_TYPE_MAP = new HashMap<>();
+        PROTO_TYPE_MAP.put(Descriptors.FieldDescriptor.Type.BYTES, TypeInfoFactory.BINARY);
+        PROTO_TYPE_MAP.put(Descriptors.FieldDescriptor.Type.STRING, TypeInfoFactory.STRING);
+        PROTO_TYPE_MAP.put(Descriptors.FieldDescriptor.Type.ENUM, TypeInfoFactory.STRING);
+        PROTO_TYPE_MAP.put(Descriptors.FieldDescriptor.Type.DOUBLE, TypeInfoFactory.DOUBLE);
+        PROTO_TYPE_MAP.put(Descriptors.FieldDescriptor.Type.FLOAT, TypeInfoFactory.FLOAT);
+        PROTO_TYPE_MAP.put(Descriptors.FieldDescriptor.Type.BOOL, TypeInfoFactory.BOOLEAN);
+        PROTO_TYPE_MAP.put(Descriptors.FieldDescriptor.Type.INT64, TypeInfoFactory.BIGINT);
+        PROTO_TYPE_MAP.put(Descriptors.FieldDescriptor.Type.UINT64, TypeInfoFactory.BIGINT);
```

Comment:
The static initialization of type mappings isn't thread-safe. Using a regular HashMap with static initialization could lead to visibility issues across threads. Consider using ImmutableMap or an enum-based approach.